packages/schedulers/lib/intervalScheduler.js

```diff
@@ -49,18 +61,33 @@ const init = async ({
       const lock = await redlock()
         .lock(`locks:${name}-scheduler`, 1000 * lockTtlSecs)
 
+      const extendLockInterval = setInterval(
+        async () =>
+          lock.extend(1000 * lockTtlSecs)
+            .then(() => { debug('extending lock') })
+            .catch(e => { console.warn('extending lock failed', e) }),
+        1000 * lockTtlSecs * 0.9
+      )
+
       debug('run started')
 
-      await runFunc({}, context)
+      if (Array.isArray(runFunc)) {
+        await Promise.each(runFunc, f => f({ runDry }, context))
+      } else if (typeof runFunc === 'function') {
+        await runFunc({ runDry }, context)
+      }
```

Comment:
IMHO would the code be more clean and resuable if runFunc has to be a function. Then one could supply `Promise.each` or what it might be for the specific usecase from external.